docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,160 @@
+    protection. Integrity of the secret can be verified by the verifier
+    using the hash in the evidence.
+  - Custom claims are NOT inspected or interpreted by OE SDK / plugins
+- Optional parameters semantics are the opposite of that of custom claims:
+  - They are TEE and evidence format specific
+  - They are interpreted and processed by OE SDK attester and verifier plugins
+    - Typically, optional parameters are provided by a verifier to an attester,
+    so the attester can generate the correct evidence that can be successfully
+    verified by the verifier
+  - They are not interpreted nor processed by attester / verifier
+  application software
+
+For SGX local attestation, the SGX target info structure generated by an
+verifier enclave for an attester enclave is a perfect fit for the optional
+parameters in the attester enclave's call to `oe_get_evidence()`:
+- For SGX local attestation, the SGX target info is generated by a verifier
+enclave and passed to an attester enclave
+- With the received target info, the attester enclave can generate an SGX
+report targeted for the verifier enclave
+- This SGX report can only be verified by the verifier enclave that provided
+by target info. Any other enclaves, either running on the local or a remote
+platform, won’t be able to verify this SGX report.
+- The target info is SGX specific, and is not parsed nor interpreted
+by the TEE-agnostic application code. It is handled by the SGX local
+attestation plugins underneath the `oe_get_evidence()` and
+`oe_verify_evidence()` API in the attester and the verifier respectively.
+
+# Attester – Verifier Security Model and TEE Agnostic Design
+
+- OE SDK is designed for attester and verifier software to be TEE agnostic,
+so that the same source code can be built for and run on different platforms
+- But owners of attesters and verifiers define their security policies,
+and the policies are not TEE agnostic. For example:
+  - The policies define which TEEs, platforms and configurations are trusted
+  to run attesters and verifiers.
+    - They also define security properties of these entities, e.g. keypairs
+    and certs, service provider ID / SPID for SGX EPID evidence, etc.
+    These security properties are required for attestation and secure
+    communication.
+  - They define which evidence formats attesters accept, and configure the
+  list to the verifiers.
+    - Every acceptable evidence format is TEE specific.
+- OE SDK provides means for the attesters and verifiers to be built and
+configured for specified TEEs and platforms, and to generate or verify
+the evidence formats according to their defined security policies.
+  - The OE SDK attestation API enables the attesters and verifiers to discover
+  and register supported plugins that are built, configured, and working
+  properly. For a specific attester or verifier instance, the supported list
+  of evidence is not required to perfectly match the list defined in the
```

Comment:
"supported list of evidence" is confusing.  Do you mean list of evidence formats?  List of evidence plugins?  Something else?

Reply:
Will clarify.